File: frontend/src/types/analysisTask.ts
import type { TaskPlan } from './taskPlan'

export interface AnalysisTask {
  id: number
  user_input: string
  topic: string
  industry?: string | null
  target_product?: string | null
  status: string
  report_depth: string
  output_language: string
  task_plan: TaskPlan
  error_message?: string | null
  created_at: string
  updated_at: string
}

File: backend/app/schemas/analysis_task.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.agent_node import AgentNodeResponse
from app.schemas.task_plan import TaskPlan


class AnalysisTaskCreateRequest(BaseModel):
    user_input: str
    task_plan: TaskPlan


class AnalysisTaskCreateResponse(BaseModel):
    task_id: int
    status: str


class AnalysisTaskResponse(BaseModel):
    id: int
    user_input: str
    topic: str
    industry: str | None
    target_product: str | None
    status: str
    report_depth: str
    output_language: str
    task_plan: TaskPlan
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime


class AnalysisTaskHistoryItem(AnalysisTaskResponse):
    nodes: list[AgentNodeResponse]


class AnalysisTaskHistoryResponse(BaseModel):
    items: list[AnalysisTaskHistoryItem]
